packages/runtime/container-runtime/src/summaryGenerator.ts

```diff
@@ -14,12 +14,28 @@ import {
 } from "./summarizerTypes";
 import { IClientSummaryWatcher } from "./summaryCollection";
 
-export const checkNotTimeout = <T>(something: T | IPromiseTimerResult | undefined): something is T => {
-    if (something === undefined) {
-        return false;
-    }
-    return (something as IPromiseTimerResult).timerResult === undefined;
+export type RaceTimerResult<T> = {
+    readonly result: IPromiseTimerResult["timerResult"];
```

Comment:
Would it be a bit more easier to follow the code if we delete IPromiseTimerResult and instead use something like
type TimerResult = "timeout" | "cancel"?
Also any chance you can add comments above IPromiseTimerResult  describing how everybody should think about what timeout and cancel mean?

Reply:
Yes, but those are common-utils changes, which require compatibility iterations. I think there are other changes/improvements to bundle within timer.ts.